extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -588,21 +588,20 @@ private QueryDefinition initializeQueryDefAndState(final Closer closer)
                                    .orElse(MSQWarnings.DEFAULT_MAX_PARSE_EXCEPTIONS_ALLOWED);
     }
 
+    WorkerStorageParameters workerStorageParameters =
+        WorkerStorageParameters.createProductionInstanceForController(context.injector(), isDurableStorageEnabled);
+
     ImmutableMap.Builder<String, Object> taskContextOverridesBuilder = ImmutableMap.builder();
     taskContextOverridesBuilder
         .put(
             MultiStageQueryContext.CTX_DURABLE_SHUFFLE_STORAGE,
             isDurableStorageEnabled
         ).put(
             MultiStageQueryContext.CTX_COMPOSED_INTERMEDIATE_SUPER_SORTER_STORAGE,
-            MultiStageQueryContext.isComposedIntermediateSuperSorterStorageEnabled(
-                task.getQuerySpec().getQuery().context()
-            )
+            isDurableStorageEnabled // If Durable Storage is enabled, then super sorter intermediate storage can be enabled.
         ).put(
             MultiStageQueryContext.CTX_INTERMEDIATE_SUPER_SORTER_STORAGE_MAX_LOCAL_BYTES,
```

Comment:
Lets update the documentation in `MultiStageQueryContext` for these params. 
Also the user facing docs for durableStorage would need to be updated.

Reply:
Updated the documentation